cg/services/orders/store_order_services/store_generic_order.py

```diff
@@ -1,99 +1,40 @@
-
-    def _validate_subject_sex(self, samples: [Of1508Sample], customer_id: str):
-        """Validate that sex is consistent with existing samples, skips samples of unknown sex
-
-        Args:
-            samples     (list[dict]):   Samples to validate
-            customer_id (str):          Customer that the samples belong to
-        Returns:
-            Nothing
-        """
-
-        sample: Of1508Sample
-        for sample in samples:
-            subject_id: str = sample.subject_id
-            if not subject_id:
-                continue
-            new_gender: str = sample.sex
-            if new_gender == "unknown":
-                continue
-
-            existing_samples: list[Sample] = self.status.get_samples_by_customer_and_subject_id(
-                customer_internal_id=customer_id, subject_id=subject_id
-            )
-            existing_sample: Sample
-            for existing_sample in existing_samples:
-                previous_gender = existing_sample.sex
-                if previous_gender == "unknown":
-                    continue
-
-                if previous_gender != new_gender:
-                    raise OrderError(
-                        f"Sample gender inconsistency for subject_id: {subject_id}: previous gender {previous_gender}, new gender {new_gender}"
-                    )
-
-    def _validate_samples_available_to_customer(
-        self, samples: list[OrderInSample], customer_id: str
-    ) -> None:
-        """Validate that the customer have access to all samples"""
-        sample: Of1508Sample
-        for sample in samples:
-            if not sample.internal_id:
-                continue
-
-            existing_sample: Sample = self.status.get_sample_by_internal_id(
-                internal_id=sample.internal_id
-            )
-
-            data_customer: Customer = self.status.get_customer_by_internal_id(
-                customer_internal_id=customer_id
-            )
-
-            if existing_sample.customer not in data_customer.collaborators:
-                raise OrderError(f"Sample not available: {sample.name}")
-
-    def _validate_case_names_are_unique(
-        self, samples: list[OrderInSample], customer_id: str
-    ) -> None:
-        """Validate that the names of all cases are unused for all samples"""
-
-        customer: Customer = self.status.get_customer_by_internal_id(
-            customer_internal_id=customer_id
-        )
-
-        sample: Of1508Sample
-        for sample in samples:
-            if self._is_rerun_of_existing_case(sample=sample):
-                continue
-            if self.status.get_case_by_name_and_customer(
-                customer=customer, case_name=sample.family_name
-            ):
-                raise OrderError(f"Case name {sample.family_name} already in use")
-
-    def submit_order(self, order: OrderIn) -> dict:
+class StoreGenericOrderService(StoreOrderService):
```

Comment:
From this naming I would expect this service to work for all order flows. I think the previous name (`CaseSubmitter)` was chosen to reflect that it was used for workflows where the customers provide data about the cases as well as the samples. It was not a perfect naming but maybe it would be possible to choose a name implying something similar?

Reply:
All of the order generate Cases, hence why I thought Case is not really meaningful. I used Generic here since the class would be used for the most use cases we have, that is all major workflows. I clarified the docstring what it is used for.

That being said since both of you think Case is clearer I will rename it.